connectors/src/connectors/notion/temporal/workflows.ts

```diff
@@ -133,21 +165,37 @@ export async function notionSyncWorkflow(
       }
     } while (cursor);
 
+    // wait for all child workflows to finish
     await Promise.all(promises);
-    await saveSuccessSyncActivity(dataSourceConfig);
-    lastSyncedPeriodTs = runTimestamp;
-    iterations += 1;
 
+    if (!isGargageCollectionRun) {
+      await saveSuccessSyncActivity(dataSourceConfig);
+      lastSyncedPeriodTs = runTimestamp;
+    } else {
+      // garbage collect the pages -- can be done non-blocking
+      void executeChild(notionGarbageCollectWorkflow.name, {
```

Comment:
the only reason this is a workflow and not an activity is because I don't want to wait for it to complete in the parent WF. The parent can be cont'd as new regardless of the status of the child